compiler/rustc_infer/src/infer/need_type_info.rs

```diff
@@ -1,8 +1,8 @@
+use crate::error_reporting::infer::TypeErrCtxt;
```

Comment:
I didn't actually move `need_type_info`, bc it has some code that accesses private `InferCtxt` fields, so I'll need to split it up more carefully later...